importer/api/__init__.py

```diff
@@ -230,6 +243,10 @@ def import_children(course, element, parent, parent_dpath):
         ]
     )
 
-    for child in element.getchildren():
-        if child.tag in DESCRIPTOR_TAGS:
-            import_children(course, child, resource, dpath)
+    # Try to protect against bad data, specifically <problem><problem>...
+    # imports. The two tags will still appear in content_xml but there will
+    # be only one resource for the outer one.
+    if not is_leaf_tag(element.tag):
```

Comment:
preference:

Make this:

```python
if is_leaf_tag(element.tag):
    continue
```

That way it's easier to read, and there's no unnecessary indentation of the important and most-commonly-executed logic.